### ISSUE TITLE
Errors when rebuilding Lazarus due to androidwidget.pas and the -Sy option

### ISSUE THREAD

**Issue by VnGNL** [jmpessoa/lazandroidmodulewizard#521]
Good time.

There is the following:
Lazarus v3.4, 
FPC 3.2.2,
x86_64-linux-gtk2,
lamw v0.8.6.4

All this has been configured using fpcupdeluxe v2.4.0e.

Everything seems to be working, but I want to fix something for myself (maybe I'll share it) using Lazarus with debugging options...

The parameters of the debug profile compiler are as follows:
Widget library LCL = gtk2
Options:
**-O- -gw3 -gl -gh -gt -Criot -Sa -Sy**

And when I run the Lazarus rebuild, 50 errors pop up modulo androidwidget.pas, of a similar type:

```
<...>
Verbose: Compiling androidwidget.pas
androidwidget.pas(5332,65) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(5360,71) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(5395,63) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(5425,63) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(5452,54) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(5478,54) Error: Incompatible types: got "^Array[0..1] Of jvalue" expected "Pjvalue"
androidwidget.pas(5506,54) Error: Incompatible types: got "^Array[0..2] Of jvalue" expected "Pjvalue"
androidwidget.pas(5543,54) Error: Incompatible types: got "^Array[0..1] Of jvalue" expected "Pjvalue"
androidwidget.pas(5572,54) Error: Incompatible types: got "^Array[0..1] Of jvalue" expected "Pjvalue"
androidwidget.pas(5604,72) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(5647,72) Error: Incompatible types: got "^Array[0..1] Of jvalue" expected "Pjvalue"
androidwidget.pas(5703,54) Error: Incompatible types: got "^Array[0..1] Of jvalue" expected "Pjvalue"
androidwidget.pas(5728,54) Error: Incompatible types: got "^Array[0..2] Of jvalue" expected "Pjvalue"
androidwidget.pas(5757,63) Error: Incompatible types: got "^Array[0..1] Of jvalue" expected "Pjvalue"
androidwidget.pas(5789,62) Error: Incompatible types: got "^Array[0..2] Of jvalue" expected "Pjvalue"
androidwidget.pas(5819,62) Error: Incompatible types: got "^Array[0..2] Of jvalue" expected "Pjvalue"
androidwidget.pas(5868,65) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(5900,54) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(5928,65) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(5956,63) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(5988,72) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(6072,72) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(6117,72) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(6153,54) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(6175,54) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(6234,54) Error: Incompatible types: got "^Array[0..1] Of jvalue" expected "Pjvalue"
androidwidget.pas(6271,54) Error: Incompatible types: got "^Array[0..1] Of jvalue" expected "Pjvalue"
androidwidget.pas(6304,54) Error: Incompatible types: got "^Array[0..6] Of jvalue" expected "Pjvalue"
androidwidget.pas(6330,65) Error: Incompatible types: got "^Array[0..1] Of jvalue" expected "Pjvalue"
androidwidget.pas(6355,65) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(6381,63) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(6446,63) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(6473,65) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(7336,82) Error: Incompatible types: got "^{Open} Array Of jvalue" expected "Pjvalue"
androidwidget.pas(7421,79) Error: Incompatible types: got "^{Open} Array Of jvalue" expected "Pjvalue"
androidwidget.pas(7428,82) Error: Incompatible types: got "^{Open} Array Of jvalue" expected "Pjvalue"
androidwidget.pas(7449,72) Error: Incompatible types: got "^{Open} Array Of jvalue" expected "Pjvalue"
androidwidget.pas(7463,83) Error: Incompatible types: got "^{Open} Array Of jvalue" expected "Pjvalue"
androidwidget.pas(7484,94) Error: Incompatible types: got "^{Open} Array Of jvalue" expected "Pjvalue"
androidwidget.pas(7539,85) Error: Incompatible types: got "^{Open} Array Of jvalue" expected "Pjvalue"
androidwidget.pas(7582,88) Error: Incompatible types: got "^{Open} Array Of jvalue" expected "Pjvalue"
androidwidget.pas(7622,77) Error: Incompatible types: got "^{Open} Array Of jvalue" expected "Pjvalue"
androidwidget.pas(7663,98) Error: Incompatible types: got "^{Open} Array Of jvalue" expected "Pjvalue"
androidwidget.pas(7714,67) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(7792,67) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(7834,67) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(7923,25) Error: Incompatible types: got "^PPointer" expected "PPointer"
androidwidget.pas(8077,67) Error: Incompatible types: got "^Array[0..1] Of jvalue" expected "Pjvalue"
androidwidget.pas(8108,55) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(8132,58) Error: Incompatible types: got "^Array[0..0] Of jvalue" expected "Pjvalue"
androidwidget.pas(8132,58) Verbose: There were 50 errors compiling module, stopping
Verbose: Compilation aborted
<...>
```

Looking at the code, I couldn't figure out what the compiler wants. The _{$mode delphi}_ directive is specified in this module. And when I remove the **-Sy** parameter from the build options, then Lazarus goes through the build with the remaining options without errors.

In short, something is wrong with the syntax.
Thank you for your attention.
